drivers/adc/ad7616/ad7616.h

```diff
@@ -235,7 +264,7 @@ int32_t ad7616_set_oversampling_ratio(struct ad7616_dev *dev,
 				      enum ad7616_osr osr);
 /* Read data in serial mode. */
 int32_t ad7616_read_data_serial(struct ad7616_dev *dev,
-				uint32_t *buf,
+				struct ad7616_conversion_result *results,
```

Comment:
When changing API, it's also expected to edit the existing project(s) that use the API, maybe as follow-up commits, so that we don't break what already works:
https://github.com/analogdevicesinc/no-OS/blob/f96a73637bbd994ffbe7281720055a619291afbf/projects/ad7616-sdz/src/ad7616_sdz.c#L144

Reply:
Yes, that's why this is still in draft. I did not do the change yet as I was not sure about how this change would be received. I found this better for the end user than "just" a buffer.
I had a discussion with @mike-bradley about maybe doing this. If you found it useless, it's trivial to revert. Let me know what you think about it

Reply:
I'm totally fine with changing it, it seems to add value. Just add a follow-up commit that edits the existing project (projects/ad7616-sdz) to compile with the new ad7616_read_data_serial() signature.

Reply:
so can you please also modify the `projects/ad7616-sdz` to properly use the new ad7616_read_data_serial ?

Reply:
Yes, I'm working on it. I have to build the setup for this project to test the changes. I will remove the draft state once I've tested the `projects/ad7616-sdz`

Reply:
Given that the default SPI Engine based project on the main branch is not stable , I think testing in hardware is not necessary before merging generic SPI support. We're in the process of debugging the SPI Engine project